Allow gptransfer to work if columns are dropped and re-added

shwu opened the pull request:
When a column is dropped and readded, Postgres assigns the column the 
next unused ordinal number (not its original one). Previously we explicitly 
compared these ordinal numbers when doing validation between tables for 
gptransfer, but this caused gptransfer to fail if ordinal numbers were not 
the same, even if the actual column order was the same. We have removed 
the ordinal number check and now compare by the order to fix this. Now it
is not necessary for the columns to have matching ordinal numbers. As long 
as the column order is the same, the transfer can complete successfully.

Authors: Karen Huddleston and Stephen Wu

pivotal-issuemaster commented:
@shwu Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=1300)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=1300) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

jimmyyih commented on gpMgmt/bin/gptransfer:
```diff
@@ -3647,8 +3647,8 @@ class GpTransfer(object):
             return False
-        for ordinal_pos in src_tbl_columns:
-            if (ordinal_pos not in dest_tbl_columns) or (src_tbl_columns[ordinal_pos] != dest_tbl_columns[ordinal_pos]):
+        for index in src_tbl_columns:
```
I suggest changing the variable name index to column unless this actually does something with indexes.

jimmyyih replied:
Replacing it with `position` is better.  Putting the words index and column near each other can imply that an index was created on top of the column and that you're only dealing with those columns instead of all table columns.

pivotal-issuemaster commented:
@shwu Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=1300)!

Chibin commented:
If possible, can we have a behave test for this?

shwu commented:
Added behave tests and addressed other PR comments.

The logic for the validation has also been cleaned up a little since we no longer care about ordinal number.
